README.md

```diff
@@ -1,2 +1,95 @@
+
+Some scenarios like in-vr navigation require entering immersive mode automatically without user gesture activation. This necessitates a standard mechanism for UAs to grant immersive sessions.
+
+## Design Goals
+
+The goal of this repository is to discuss and design a simple API for UAs to grant immersive sessions without an explicit user activation in specific scenarios.
+
+## Non Goals
+
+Define how a Virtual Reality or Augmented Reality browser would work [in alignment with the WebXR spec](https://github.com/immersive-web/webxr/blob/master/explainer.md#non-goals)
+
+## User Scenarios
+
+1. Immersive navigation, allowing UAs to implement seamless transitions between interlinked VR and AR content. Use cases include:
+
+    - Games that contains links to other games. Or links to friends' games.
+    - Personal blogs that link to affiliated VR sites or your friends blogs.
+    - Educational websites for a class, that link to the school's portal or other resources.
+    - User generated content linking to one another. Like Rec Room, VR Chat or Super Craft.
+    - Sponsored links. You have a VR world about photography and filmmaking that links to Photorama, B&H, GoPro's VR portals.
+    - Content portals in general (directories of games, immersive videos, collections).
+    - Personal storefronts that link to products hosted by eBay VR, Amazon VR. One could also easily incorporate product 3D models hosted externally in services the like of Sketchfab or Google Blocks.
+
+2. In an AR browser, virtual objects might be fetched from different URLs and rendered in physical context. Content is expected to always present at load time.
+
+3. Open in immersive mode. 2D windowed browsers might offer a way to open a link directly immersive mode, similar to "open in new tab" or "open in new page" actions offered today.
+
+4. Intents, deep linking, PWA, kiosk mode. Native aplications might request the browser to open content directly in immersive mode.
+
+5. Performance implications of single page applications. Having different web pages load is sort of beneficial for not having to deal with offloading / onloading new models and environments or making your own interstitial. You can avoid frame drops by letting the browser handle the transition, and VR web pages load quick enough.
+
+## API Proposal
+
+UAs fire a `sessiongranted` event that allows content to enter immersive mode without user gesture activation:
+
+```
+navigator.xr.addEventListener('sessiongranted', function (evt) {
+   // One could check for the type of session granted.
+   // Events notifies of session creation after navigation, UA action, or requestSession.
+   // The session object is provided as part of this event.
+   if (evt.session.mode === 'immersive-vr') {
+      // set up app state for immersive vr, if that's what the app wants
+   } else {
+      // notify user that this app only works in immersive vr mode, if desired
```

Comment:
How do we "reject" a session?  Do we not, in this case?  Do we at least require/suggest that apps display "something" that indicates this situation?

Reply:
Not sure if we need explicit rejection. WebVR `vrdisplayactivate` event handler times out. `requestPresent` is not subject to user gesture for limited time and privilege revoked if not used.